compiler/pipes/fix-array-access.h

```diff
@@ -0,0 +1,18 @@
+// Compiler for PHP (aka KPHP)
+// Copyright (c) 2024 LLC «V Kontakte»
+// Distributed under the GPL v3 License, see LICENSE.notice.txt
+
+#pragma once
+
+#include "compiler/function-pass.h"
+
+class FixArrayAccessPass final : public FunctionPassBase {
```

Comment:
I'd suggest to change the name of the pass. For example, `ArrayAccessTransformPass`